task(shared): Fix docker network error with otel

dschom opened the pull request:
## Because

- When starting tracing with open telemetry the network name would sometimes collide and result in error.

## This pull request

- Sets an 'fxa' network name
- Applies this network to services
- Allows services to communicate by their names

biancadanforth reviewed (approved):
I can confirm this fixes the issue of starting up the tracing services (`otel-collector` and `jaeger`) locally.

It would be great to understand why there was already an `otel` network in Docker and what it's used for.

(Previous error from both `otel-collector` and `jaeger` error logs:)
```
Error response from daemon: network with name otel already exists
docker: Error response from daemon: network otel is ambiguous (2 matches found on name).
```

dschom commented:
> It would be great to understand why there was already an `otel` network in Docker and what it's used for.

Thanks for the review! 

By the way, both the jeager.sh and otel-collector.sh files attempted to create the network without first checking for its existence. This is why it existed twice and is also what this PR addresses. 

I added this network so that the otel-collector and the Jaeger service could communicate using their docker names as hostnames on the docker network. In general for a local development stack like this, it is convenient for all services to be on the same docker network, which is why I've gone ahead and added an `fxa` network that all our docker services can be part of.

biancadanforth commented:
That makes sense; thanks for the explanation. The ol' "create if not exists".
